File: generators/starTrekFerengi.py
__all__ = ['starTrekFerengi']

# Don't look below, you will not understand this Python code :) I don't.

from js2py.pyjs import *
# setting scope
var = Scope( JS_BUILTINS )
set_global_object(var)

# Code follows:
var.registers(['nm1', 'nm2', 'nm3', 'nm4', 'nameGen'])
@Js
def PyJsHoisted_nameGen_(type, this, arguments, var=var):
    var = Scope({'arguments':arguments, 'this':this, 'type':type}, var)
    var.registers(['tp', 'result', 'type'])
    var.put('tp', var.get('type'))
    var.put('result', Js([]))
    #for JS loop
    var.put('i', Js(0.0))
    while (var.get('i')<Js(10.0)):
        try:
            if PyJsStrictEq(var.get('tp'),Js(1.0)):
                var.put('rnd', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm3').get('length'))))
                var.put('rnd2', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm4').get('length'))))
                var.put('names', (var.get('nm3').get(var.get('rnd'))+var.get('nm4').get(var.get('rnd2'))))
            else:
                var.put('rnd', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm1').get('length'))))
                var.put('rnd2', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm2').get('length'))))
                var.put('names', (var.get('nm1').get(var.get('rnd'))+var.get('nm2').get(var.get('rnd2'))))
            var.get('result').callprop('push', var.get('names'))
        finally:
                (var.put('i',Js(var.get('i').to_number())+Js(1))-Js(1))
    pass
    return var.get('result')
PyJsHoisted_nameGen_.func_name = 'nameGen'
var.put('nameGen', PyJsHoisted_nameGen_)
var.put('nm1', Js([Js('Aba'), Js('Adre'), Js('Aga'), Js('Arga'), Js('Arri'), Js('Ba'), Js('Be'), Js('Belo'), Js('Beri'), Js('Bo'), Js('Bra'), Js('Bre'), Js('Broi'), Js('Bru'), Js('Da'), Js('De'), Js('Di'), Js('Do'), Js('Dra'), Js('Dre'), Js('Droi'), Js('Fa'), Js('Fale'), Js('Fare'), Js('Fna'), Js('Fra'), Js('Fre'), Js('Fri'), Js('Froo'), Js('Ga'), Js('Gai'), Js('Ge'), Js('Gegi'), Js('Gi'), Js('Gla'), Js('Gna'), Js('Go'), Js('Gora'), Js('Gra'), Js('Gri'), Js('Groo'), Js('Ha'), Js('Hai'), Js('Ho'), Js('Hoe'), Js('Hora'), Js('Hra'), Js('Iga'), Js('Ige'), Js('Ira'), Js('Iro'), Js('Ita'), Js('Ka'), Js('Kara'), Js('Kay'), Js('Kaza'), Js('Ke'), Js('Ki'), Js('Kola'), Js('Kra'), Js('Kre'), Js('Kri'), Js('La'), Js('Le'), Js('Lera'), Js('Lete'), Js('Li'), Js('Lo'), Js('Lu'), Js('Luri'), Js('Ma'), Js('Mala'), Js('Maza'), Js('Mo'), Js('Mona'), Js('Mora'), Js('Mu'), Js('Na'), Js('Nala'), Js('Nazra'), Js('Nera'), Js('Ni'), Js('Nibo'), Js('Nilra'), Js('No'), Js('Nu'), Js('Ola'), Js('Ona'), Js('Ora'), Js('Orda'), Js('Orpa'), Js('Oza'), Js('Pa'), Js('Pela'), Js('Pera'), Js('Pi'), Js('Ple'), Js('Pra'), Js('Ra'), Js('Rate'), Js('Raza'), Js('Re'), Js('Rha'), Js('Ro'), Js('Rotto'), Js('Ruta'), Js('Sa'), Js('Smee'), Js('So'), Js('Solo'), Js('Sova'), Js('Sra'), Js('Sto'), Js('Sura'), Js('Ta'), Js('Tega'), Js('To'), Js('Torro'), Js('Toza'), Js('Tra'), Js('Tro'), Js('Tu'), Js('Turo'), Js('Tye'), Js('Ubo'), Js('Ugro'), Js('Ulda'), Js('Uli'), Js('Ura'), Js('Uro'), Js('Ya'), Js('Yaza'), Js('Ye'), Js('Yna'), Js('Yora'), Js('Za'), Js('Ze'), Js('Zira'), Js('Zra'), Js('Zyla')]))
var.put('nm2', Js([Js('ba'), Js('bac'), Js('bar'), Js('bor'), Js('c'), Js('car'), Js('ck'), Js('ctor'), Js('d'), Js('dac'), Js('dok'), Js('dor'), Js('g'), Js('ga'), Js('gg'), Js('ggie'), Js('gis'), Js('go'), Js('k'), Js('ka'), Js('kag'), Js('kor'), Js('l'), Js('la'), Js('ldar'), Js('lis'), Js('lok'), Js('lva'), Js('m'), Js('mag'), Js('mp'), Js('mra'), Js('n'), Js('nk'), Js('no'), Js('nog'), Js('nt'), Js('nzo'), Js('p'), Js('pak'), Js('pax'), Js('pum'), Js('rad'), Js('rbo'), Js('rik'), Js('rin'), Js('ron'), Js('rot'), Js('rpax'), Js('rr'), Js('rrot'), Js('rta'), Js('s'), Js('sax'), Js('sh'), Js('ss'), Js('t'), Js('ta'), Js('tax'), Js('tek'), Js('va'), Js('vag'), Js('vak'), Js('var'), Js('xa'), Js('xac'), Js('xon'), Js('xor'), Js('za'), Js('zar'), Js('zig'), Js('zoc')]))
var.put('nm3', Js([Js('A'), Js('Ai'), Js('Ali'), Js('Ana'), Js('Ara'), Js('Be'), Js('Bee'), Js('Bela'), Js('Bia'), Js('Bira'), Js('Bli'), Js('Bo'), Js('Bona'), Js('Bre'), Js('Da'), Js('De'), Js('Dera'), Js('Dia'), Js('Dina'), Js('Do'), Js('Dri'), Js('Fa'), Js('Fe'), Js('Fena'), Js('Fesa'), Js('Fhi'), Js('Fi'), Js('Fia'), Js('Fira'), Js('Ga'), Js('Gase'), Js('Ge'), Js('Gela'), Js('Geli'), Js('Ghe'), Js('Gi'), Js('Gine'), Js('Gira'), Js('Giya'), Js('Gre'), Js('Gri'), Js('Hela'), Js('Hena'), Js('Hi'), Js('Hia'), Js('Hy'), Js('Hyra'), Js('I'), Js('Ina'), Js('Ira'), Js('Iro'), Js('Isa'), Js('Ka'), Js('Ke'), Js('Keha'), Js('Kella'), Js('Kena'), Js('Ki'), Js('Kia'), Js('Kisa'), Js('Kora'), Js('Kre'), Js('La'), Js('Le'), Js('Lera'), Js('Leya'), Js('Li'), Js('Lina'), Js('Losa'), Js('Ma'), Js('Mane'), Js('Me'), Js('Mero'), Js('Mi'), Js('Mia'), Js('Mo'), Js('Nala'), Js('Ne'), Js('Nema'), Js('Nemo'), Js('Nia'), Js('Nira'), Js('No'), Js('Noa'), Js('Nosi'), Js('O'), Js('Ole'), Js('One'), Js('Ora'), Js('Ori'), Js('Oza'), Js('Pa'), Js('Pe'), Js('Peha'), Js('Phi'), Js('Poli'), Js('Pri'), Js('Re'), Js('Rema'), Js('Rena'), Js('Ri'), Js('Ria'), Js('Rika'), Js('Rosa'), Js('Roxe'), Js('Sa'), Js('Sela'), Js('Seno'), Js('Sha'), Js('Si'), Js('Sira'), Js('Sra'), Js('Sti'), Js('Ta'), Js('Te'), Js('Tewa'), Js('Teza'), Js('Thi'), Js('Tia'), Js('Tila'), Js('Tiro'), Js('To'), Js('Tona'), Js('U'), Js('Ula'), Js('Uni'), Js('Ure'), Js('Uye'), Js('Uza'), Js('Ya'), Js('Yala'), Js('Ye'), Js('Yne'), Js('Yra'), Js('Zane'), Js('Ze'), Js('Zeri'), Js('Zia')]))
var.put('nm4', Js([Js('ba'), Js('bel'), Js('bis'), Js('bora'), Js('ca'), Js('cera'), Js('cine'), Js('cis'), Js('d'), Js('de'), Js('del'), Js('dis'), Js('ga'), Js('gel'), Js('gela'), Js('gena'), Js('gis'), Js('gora'), Js('k'), Js('kera'), Js('kia'), Js('kis'), Js('l'), Js('le'), Js('lea'), Js('lera'), Js('lina'), Js('ll'), Js('me'), Js('mera'), Js('mis'), Js('miya'), Js('n'), Js('ne'), Js('nia'), Js('nka'), Js('nni'), Js('no'), Js('pe'), Js('phi'), Js('pia'), Js('pora'), Js('r'), Js('re'), Js('res'), Js('reya'), Js('ri'), Js('ris'), Js('rles'), Js('rni'), Js('rona'), Js('rra'), Js('s'), Js('se'), Js('sh'), Js('sha'), Js('shi'), Js('shka'), Js('si'), Js('ta'), Js('the'), Js('ti'), Js('tis'), Js('va'), Js('vena'), Js('vil'), Js('vira'), Js('xaia'), Js('xen'), Js('xera'), Js('xi'), Js('zel'), Js('zenna'), Js('zera')]))
pass
pass


# Add lib to the module scope
starTrekFerengi = var.to_python()

File: js/starTrekFerengi.js
var nm1 = ["Aba","Adre","Aga","Arga","Arri","Ba","Be","Belo","Beri","Bo","Bra","Bre","Broi","Bru","Da","De","Di","Do","Dra","Dre","Droi","Fa","Fale","Fare","Fna","Fra","Fre","Fri","Froo","Ga","Gai","Ge","Gegi","Gi","Gla","Gna","Go","Gora","Gra","Gri","Groo","Ha","Hai","Ho","Hoe","Hora","Hra","Iga","Ige","Ira","Iro","Ita","Ka","Kara","Kay","Kaza","Ke","Ki","Kola","Kra","Kre","Kri","La","Le","Lera","Lete","Li","Lo","Lu","Luri","Ma","Mala","Maza","Mo","Mona","Mora","Mu","Na","Nala","Nazra","Nera","Ni","Nibo","Nilra","No","Nu","Ola","Ona","Ora","Orda","Orpa","Oza","Pa","Pela","Pera","Pi","Ple","Pra","Ra","Rate","Raza","Re","Rha","Ro","Rotto","Ruta","Sa","Smee","So","Solo","Sova","Sra","Sto","Sura","Ta","Tega","To","Torro","Toza","Tra","Tro","Tu","Turo","Tye","Ubo","Ugro","Ulda","Uli","Ura","Uro","Ya","Yaza","Ye","Yna","Yora","Za","Ze","Zira","Zra","Zyla"];
var nm2 = ["ba","bac","bar","bor","c","car","ck","ctor","d","dac","dok","dor","g","ga","gg","ggie","gis","go","k","ka","kag","kor","l","la","ldar","lis","lok","lva","m","mag","mp","mra","n","nk","no","nog","nt","nzo","p","pak","pax","pum","rad","rbo","rik","rin","ron","rot","rpax","rr","rrot","rta","s","sax","sh","ss","t","ta","tax","tek","va","vag","vak","var","xa","xac","xon","xor","za","zar","zig","zoc"];
var nm3 = ["A","Ai","Ali","Ana","Ara","Be","Bee","Bela","Bia","Bira","Bli","Bo","Bona","Bre","Da","De","Dera","Dia","Dina","Do","Dri","Fa","Fe","Fena","Fesa","Fhi","Fi","Fia","Fira","Ga","Gase","Ge","Gela","Geli","Ghe","Gi","Gine","Gira","Giya","Gre","Gri","Hela","Hena","Hi","Hia","Hy","Hyra","I","Ina","Ira","Iro","Isa","Ka","Ke","Keha","Kella","Kena","Ki","Kia","Kisa","Kora","Kre","La","Le","Lera","Leya","Li","Lina","Losa","Ma","Mane","Me","Mero","Mi","Mia","Mo","Nala","Ne","Nema","Nemo","Nia","Nira","No","Noa","Nosi","O","Ole","One","Ora","Ori","Oza","Pa","Pe","Peha","Phi","Poli","Pri","Re","Rema","Rena","Ri","Ria","Rika","Rosa","Roxe","Sa","Sela","Seno","Sha","Si","Sira","Sra","Sti","Ta","Te","Tewa","Teza","Thi","Tia","Tila","Tiro","To","Tona","U","Ula","Uni","Ure","Uye","Uza","Ya","Yala","Ye","Yne","Yra","Zane","Ze","Zeri","Zia"];
var nm4 = ["ba","bel","bis","bora","ca","cera","cine","cis","d","de","del","dis","ga","gel","gela","gena","gis","gora","k","kera","kia","kis","l","le","lea","lera","lina","ll","me","mera","mis","miya","n","ne","nia","nka","nni","no","pe","phi","pia","pora","r","re","res","reya","ri","ris","rles","rni","rona","rra","s","se","sh","sha","shi","shka","si","ta","the","ti","tis","va","vena","vil","vira","xaia","xen","xera","xi","zel","zenna","zera"];
var br = "";

function nameGen(type){
	$('#placeholder').css('textTransform', 'capitalize');
	var tp = type;
	var element = document.createElement("div");
	element.setAttribute("id", "result");
	
	for(i = 0; i < 10; i++){
		if(tp === 1){
			rnd = Math.floor(Math.random() * nm3.length);
			rnd2 = Math.floor(Math.random() * nm4.length);
			names = nm3[rnd] + nm4[rnd2];
		}else{
			rnd = Math.floor(Math.random() * nm1.length);
			rnd2 = Math.floor(Math.random() * nm2.length);
			names = nm1[rnd] + nm2[rnd2];
		}
		br = document.createElement('br');	
		element.appendChild(document.createTextNode(names));
		element.appendChild(br);
	}
	if(document.getElementById("result")){
		document.getElementById("placeholder").removeChild(document.getElementById("result"));
	}		
	document.getElementById("placeholder").appendChild(element);
}